### DISCUSSION TITLE
Bind SNZB-01 single/double long press to different devices

### DISCUSSION THREAD

**Discussion by SynAckFin** [Koenkk/zigbee2mqtt#15820] [Devices]
I have a Sonoff SNZB-01 (button1) and 3 x Tuya TS00001 relays (relay1/2/3).
I would like to directly bind single press to relay1, double press to relay2 and long press to relay3 so that the relays are toggled individually.

At the moment, when I bind button1 to relay1 a single press toggles the relay, a double turns it on and a long press turns it off. I only want relay1 to react to a single press and to be able to bind a double press to relay2 etc.

Is it possible to do this and if so how?

I can do all this via MQTT but having had the server fail and being unable to control anything I would like to make the button/relay control as  independent as possible.

The button1 clusters looks like this:
<code>
Button1
+--Endpoint1
.  |
.  +--Output Clusters
.  |  +--genOnOff
.  |  +--genIdentity
.  |
.  +--Input clusters
.     +--genBasic
.     +--genIdentity
.     +--genPowercfg
</code>
The relay1 Clusters look like this:
<code>
Relay1
+--Endpoint1
|  |
|  +--Output Clusters
|  |  +--genOta
|  |  +--genTime
|  |
|  +--Input clusters
|     +--genBasic
|     +--genIdentity
|     +--genGroups
|     +--genScenes
|     +--genOnOff
|     +--57344
|     +--manuSpecificTuya_3
| 
+--Endpoint 242
.  |
.  +--Output Clusters
.     +--greenPower
</code>

**Comment by pax0r** [comment]
Hello, have you managed to solve this? I have this button and I want to controll diffrent devices with long press and normal. I was able to bind normal single click, but not long press or double click.

**Comment by SynAckFin** [comment]
I haven't found any way of doing this other than via MQTT.
